packages/reports/addon/components/power-select-collection-options.ts

```diff
@@ -4,13 +4,26 @@
  *
  * An ember-power-select options component that uses ember-collection
  */
-import { alias } from '@ember/object/computed';
 import { A } from '@ember/array';
-import { action, setProperties, computed } from '@ember/object';
+import { action, computed } from '@ember/object';
 import Options from 'ember-power-select/components/power-select/options';
 import { groupBy } from 'lodash-es';
 
+type ItemOptions = Options['args']['options'];
+type Args = Options['args'] & {
+  extra?: {
+    allowClear?: boolean;
+    groupKey?: string;
+    sortKey?: string;
+    sortFn?: (options: ItemOptions) => ItemOptions;
+  };
+};
+
+export type IndexedOptions<Option = ItemOptions[number]> = { option: Option; idx: number };
+
 export default class PowerSelectCollectionOptions extends Options {
+  declare args: Args;
```

Comment:
Interesting, does something like this not work 
```
PowerSelectCollectionOptions<Args> extends Options
```
```
(PowerSelectCollectionOptions extends Options)<Args>

```

Reply:
Nope, easiest way is to override it like this unfortunately